bedboss/bedboss.py

```diff
@@ -0,0 +1,193 @@
+        )
+        # return None
+    osm_path = os.path.join(OPEN_SIGNAL_FOLDER, osm_name)
+    if not os.path.exists(osm_path):
+        if not os.path.exists(OPEN_SIGNAL_FOLDER):
+            os.makedirs(OPEN_SIGNAL_FOLDER)
+        download_file(url=f"{OPEN_SIGNAL_URL}{osm_name}", path=osm_path)
+    return osm_path
+
+
+def run_all(
+    sample_name: str,
+    input_file: str,
+    input_type: str,
+    outfolder: str,
+    genome: str,
+    bedbase_config: str,
+    rfg_config: str = None,
+    narrowpeak: bool = False,
+    check_qc: bool = True,
+    standard_chrom: bool = False,
+    chrom_sizes: str = None,
+    open_signal_matrix: str = None,
+    ensdb: str = None,
+    sample_yaml: str = None,
+    just_db_commit: bool = False,
+    no_db_commit: bool = False,
+    force_overwrite: bool = False,
+    pm: pypiper.PipelineManager = None,
+    **kwargs,
+) -> NoReturn:
+    """
+    Run bedboss: bedmaker, bedqc and bedstat.
+    :param sample_name: Sample name [required]
+    :param input_file: Input file [required]
+    :param input_type: Input type [required] options: (bigwig|bedgraph|bed|bigbed|wig)
+    :param outfolder: Folder, where output should be saved  [required]
+    :param genome: genome_assembly of the sample. [required] options: (hg19, hg38) #TODO: add more
+    :param bedbase_config: a path to the bedbase configuration file. [required] #TODO: add example
+    :param open_signal_matrix: a full path to the openSignalMatrix required for the tissue [optional]
+    :param rfg_config: file path to the genome config file [optional]
+    :param narrowpeak: whether the regions are narrow
+        (transcription factor implies narrow, histone mark implies broad peaks) [optional]
+    :param check_qc: set True to run quality control during badmaking [optional] (default: True)
+    :param standard_chrom: Standardize chromosome names. [optional] (Default: False)
+    :param chrom_sizes: a full path to the chrom.sizes required for the bedtobigbed conversion [optional]
+    :param sample_yaml: a yaml config file with sample attributes to pass on MORE METADATA into the database [optional]
+    :param ensdb: a full path to the ensdb gtf file required for genomes not in GDdata [optional]
+        (basically genomes that's not in GDdata)
+    :param just_db_commit: whether just to commit the JSON to the database (default: False)
+    :param force_overwrite: force overwrite analysis
+    :param no_db_commit: whether the JSON commit to the database should be skipped (default: False)
+    :param pm: pypiper object
+    :return: NoReturn
+    """
+    _LOGGER.warning(f"Unused arguments: {kwargs}")
+    file_name = extract_file_name(input_file)
+    genome = standardize_genome_name(genome)
+
+    # find/download open signal matrix
+    if not open_signal_matrix:
+        open_signal_matrix = get_osm_path(genome)
+    else:
+        if not os.path.exists(open_signal_matrix):
+            open_signal_matrix = get_osm_path(genome)
```

Comment:
I think this could be simplified with an `or` conditional. maybe:

```
   if (not open_signal_matrix) or (not os.path.exists(open_signal_matrix)):
        open_signal_matrix = get_osm_path(genome)
```

?